File: backend/src/main/java/com/jaee/entity/BuilderOption.java
package com.jaee.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "builder_options", uniqueConstraints = @UniqueConstraint(columnNames = {"builder_type", "option_type", "option_key"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BuilderOption {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "builder_type", nullable = false, length = 20)
    private String builderType;  // "CANDLE" or "HAMPER"

    @Column(name = "option_type", nullable = false, length = 30)
    private String optionType;   // SIZE, WAX, SCENT, COLOR, CONTAINER, ITEM, WRAPPING, OCCASION, COLOR_THEME

    @Column(name = "option_key", nullable = false, length = 50)
    private String optionKey;    // "soy", "jar", "lavender", etc.

    @Column(nullable = false, length = 100)
    private String label;

    private String description;

    @Column(length = 10)
    private String emoji;

    @Column(name = "hex_color", length = 7)
    private String hexColor;

    @Column(name = "colors_json", length = 255)
    private String colorsJson;   // for color themes: JSON array like ["#B4617B","#D4A843","#F2E3E8"]

    @Column(name = "base_price", precision = 10, scale = 2)
    @Builder.Default
    private BigDecimal basePrice = BigDecimal.ZERO;

    @Column(precision = 10, scale = 2)
    @Builder.Default
    private BigDecimal surcharge = BigDecimal.ZERO;

    @Builder.Default
    private Boolean active = true;

    @Column(name = "display_order")
    @Builder.Default
    private Integer displayOrder = 0;
}

File: mobile/lib/models/builder_option.dart
class BuilderOption {
  final int id;
  final String builderType;
  final String optionType;
  final String optionKey;
  final String label;
  final double? basePrice;
  final double? surcharge;
  final bool active;
  final int displayOrder;

  const BuilderOption({
    required this.id,
    required this.builderType,
    required this.optionType,
    required this.optionKey,
    required this.label,
    this.basePrice,
    this.surcharge,
    required this.active,
    required this.displayOrder,
  });

  factory BuilderOption.fromJson(Map<String, dynamic> json) {
    return BuilderOption(
      id: json['id'] as int,
      builderType: json['builderType'] as String,
      optionType: json['optionType'] as String,
      optionKey: json['optionKey'] as String? ?? '',
      label: json['label'] as String,
      basePrice: (json['basePrice'] as num?)?.toDouble(),
      surcharge: (json['surcharge'] as num?)?.toDouble(),
      active: json['active'] as bool? ?? true,
      displayOrder: json['displayOrder'] as int? ?? 0,
    );
  }
}
